server/channels/api4/report.go

```diff
@@ -85,3 +88,46 @@ func getUsersForReporting(c *Context, w http.ResponseWriter, r *http.Request) {
 		c.Logger.Warn("Error writing response", mlog.Err(jsonErr))
 	}
 }
+
+func startUsersBatchExport(c *Context, w http.ResponseWriter, r *http.Request) {
+	if !(c.IsSystemAdmin() && c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionSysconsoleReadUserManagementUsers)) {
+		c.SetPermissionError(model.PermissionSysconsoleReadUserManagementUsers)
+		return
+	}
+
+	if err := c.App.StartUsersBatchExport(c.AppContext, r.URL.Query().Get("date_range")); err != nil {
+		c.Err = err
+		return
+	}
+
+	ReturnStatusOK(w)
+}
+
+func retrieveBatchReportFile(c *Context, w http.ResponseWriter, r *http.Request) {
+	if !(c.IsSystemAdmin() && c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionSysconsoleReadUserManagementUsers)) {
+		c.SetPermissionError(model.PermissionSysconsoleReadUserManagementUsers)
+		return
+	}
+
+	reportId := c.Params.ReportId
+	if reportId == "" || !model.IsValidId(reportId) {
+		c.Err = model.NewAppError("retrieveBatchReportFile", "api.retrieveBatchReportFile.invalid_report_id", nil, "", http.StatusBadRequest)
+		return
+	}
+
+	format := r.URL.Query().Get("format")
+	if !model.IsValidReportExportFormat(format) {
+		c.Err = model.NewAppError("retrieveBatchReportFile", "api.retrieveBatchReportFile.invalid_format", nil, "", http.StatusBadRequest)
+		return
+	}
+
+	file, name, err := c.App.RetrieveBatchReport(reportId, format)
+	if err != nil {
+		c.Err = err
+		return
+	}
+	defer file.Close()
+
+	w.Header().Set("Content-Type", "text/csv") // TODO: other formats
```

Comment:
This was a TODO for me, more to mark that we would be setting something else for different formats.
I'm going to remove the TODO but I'll leave the comment so that we know what to change when implementing other formats.

Reply:
>  I'm going to remove the TODO but I'll leave the comment so that we know what to change when implementing other formats.

nit: It really feels like it would be obvious what to chagne when implementing other formats, so I'd be inclined to narrow in on CSV vs. future proofing comments.

Reply:
I'll remove the comment altogether then if you feel it's getting in the way.